OpenIGTLinkIF/MRML/vtkMRMLIGTLConnectorNode.cxx

```diff
@@ -832,10 +905,16 @@ void vtkMRMLIGTLConnectorNode::ProcessIOConnectorEvents(vtkObject *caller, unsig
       }
     else if (event==modifiedDevice->CommandResponseReceivedEvent)
       {
-      // TODO: It is inconsistent that command events are invoked with vtkSlicerOpenIGTLinkIFCommand attached
-      // and command response events are invoked with igtlio::Device attached.
-      // Probably command response event should be invoked with vtkSlicerOpenIGTLinkIFCommand attached.
-      this->InvokeEvent(mrmlEvent, modifiedDevice);
+      // Invoke command response event with corresponding vtkSlicerOpenIGTLinkIFCommand if found, otherwise, invoke with device
```

Comment:
I understand how this is an inconsistent behavior, hopefully this addresses it somewhat.
The remaining issue with this would be that if a command is invoked using the SendCommand function that does not take a vtkSlicerOpenIGTLinkIFCommand, then there would be no vtkSlicerOpenIGTLinkIFCommand to respond to.
Does this change make sense (Invoke with vtkSlicerOpenIGTLinkIFCommand if found, device otherwise?)